crates/engine/src/resolver/graphql/deserialize/errors.rs

```diff
@@ -86,7 +64,16 @@ where
                 error
             })
             .collect();
-        self.0.response().set_subgraph_errors(errors);
+        self.response.borrow_mut().set_subgraph_errors(errors);
         Ok(errors_count)
     }
 }
+
+#[derive(serde::Deserialize)]
+struct SubgraphGraphqlError {
+    message: String,
+    #[serde(default)]
+    path: serde_json::Value,
+    #[serde(default)]
+    extensions: serde_json::Value,
```

Comment:
Nitpick: can't we assume this is a `Map<String, serde_json::Value>`?

Reply:
We could, but we propagate it as is, so this avoids dealing with invalid formats. It'll be up to the clients to deal with it if this happens.

Reply:
makes sense :+1: